### PR TITLE
New modules + allowance management

### PR DESCRIPTION
**Pull request by u-hubar**
- Added assertion/network modules
- Allowance management function
- Extended demo.py

### CODE REVIEW

File: dkg/network.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+from dkg.manager import DefaultRequestManager
+from dkg.method import Method
+from dkg.module import Module
+from dkg.types import DataHexStr
+from dkg.utils.blockchain_request import BlockchainRequest
+from dkg.utils.node_request import NodeRequest
+
+
+class Network(Module):
+    DEFAULT_HASH_FUNCTION_ID = 1
```
Comment (u-hubar) [inline dkg/network.py:28]: Move to the constants

File: dkg/network.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    def __init__(self, manager: DefaultRequestManager):
+        self.manager = manager
+
+    _chain_id = Method(BlockchainRequest.chain_id)
+    _get_asset_storage_address = Method(BlockchainRequest.get_asset_storage_address)
+
+    _get_bid_suggestion = Method(NodeRequest.bid_suggestion)
+
+    def get_bid_suggestion(
+        self, merkle_root: DataHexStr, size_in_bytes: int, epochs_number: int,
```
Comment (u-hubar) [inline dkg/network.py:39]:
```suggestion
        self, public_assertion_id: DataHexStr, size_in_bytes: int, epochs_number: int,
```

File: dkg/assertion.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+
+class Assertion(Module):
+    def __init__(self, manager: DefaultRequestManager):
+        self.manager = manager
+
+    def format_graph(self, content: dict[Literal["public", "private"], JSONLD]):
+        return format_content(content)
+
+    def get_merkle_root(
```
Comment (u-hubar) [inline dkg/assertion.py:35]:
```suggestion
    def get_public_assertion_id(
```

File: dkg/asset.py [hunk 4]
```diff
@@ -55,12 +55,53 @@ class ContentAsset(Module):
...
+
+        return int(
+            self._get_current_allowance(
+                self.manager.blockchain_provider.account.address, spender
+            )
+        )
+
     _increase_allowance = Method(BlockchainRequest.increase_allowance)
+
+    def increase_allowance(
```
Comment (u-hubar) [inline dkg/asset.py:73]:
```suggestion
    def set_allowance(
```

File: examples/demo.py [hunk 5]
```diff
@@ -46,25 +64,55 @@ def divider():
...
-    },
-    2,
-)
+formatted_assertions = dkg.assertion.format_graph(content)
+print("======================== ASSET FORMATTED")
+print(formatted_assertions)
+
+divider()
+
+merkle_root = dkg.assertion.get_merkle_root(content)
```
Comment (u-hubar) [inline examples/demo.py:73]:
```suggestion
merkle_root = dkg.assertion.get_public_assertion_id(content)
```

**Review by NZT48 — APPROVED** [review]

**Review by NZT48 — DISMISSED** [review]

**Review by milosstanisavljevic — APPROVED** [review]

**Comment by milosstanisavljevic** [comment]
we should change python version in readme aswell

**Review by NZT48 — APPROVED** [review]

**Pull request merged**